### ISSUE TITLE
Generate alias() for all elements in override_targets 

### ISSUE THREAD

**Issue by t-soliduslink** [bazelbuild/rules_jvm_external#266]
It seems that an `alias(...)` is only generated for those elements that are actually part of the transitive dependency graph of the specified artifacts. If a specified element is not actually directly or indirectly used by one of the artifacts, it is simply discarded.

I would like to ask to generate the `alias(...)` for all the specified elements in override_targets - unconditionally and with public visibility.

This would allow me to use a consistent naming scheme of `@maven//:foo_bar_baz` for Java-related dependencies.

Concrete example:

Something else is already pulling in `@com_google_protobuf`, and I would like to specify

```
override_targets = {
         "com.google.protobuf:protobuf-java" : "@com_google_protobuf//:protobuf_java,
          ...
}
```

even though none of my specified artifacts happen to actually depend on protobuf.

Right now, I can use `@maven//:foo_bar_baz` for all of my Java-related dependencies except for protobufs where I need to specify `@com_google_protobuf//:protobuf_java`.

**Comment by jin** [comment]
This is definitely an oversight. Happy to take a PR (or I can implement this when I find the bandwidth)

**Comment by jin** [comment]
On further thought, I think we should **not** add this. It adds an indirection abstraction and a scope creep into `maven_install`, by allowing arbitrary targets to be added into the `@maven` BUILD file, making it go out of sync with `maven_install.json` or Coursier's own dep graph.

**Comment by t-soliduslink** [comment]
Could you elaborate, please, on your thought process?

It seems to me that I want to have `"com.google.protobuf:protobuf-java" : "@com_google_protobuf//:protobuf_java` in my `override_targets` even though it is a NO-OP right now: to ensure that if an artifact gets added later on that /does/ happen to depend on protobuf, that it will use the same version that is pulled in elsewhere.

And it is very surprising that only some of the elements in `override_targets` can actually be referenced under `@maven`. How is one supposed to know by casually looking?

How do you / does your team handle this issue? have you not run into it?

**Comment by jin** [comment]
The idea of `override_targets` is to only override generated `@maven//:<target>` in `@maven//:all`. `@maven//:all` consists of the transitive dependency graph of `maven_install.artifacts`. 

If you want to override `com.google.protobuf:protobuf-java` in `override_artifacts`, add that artifact into `maven_install.artifacts` as well. The final generated BUILD file in `@maven//:BUILD` will contain your override as expected.

**Comment by t-soliduslink** [comment]
Thanks for the suggestion. I just tried to explicitly specify protobuf as an `artefacts`. However, I ran into Issue #273.

One thing that I do not like about explicitly adding it to `artefacts` is that an entry there requires specifying a version. Which needs to be a valid version that exists in the maven repo (so I cannot just make it something that signifies that the entry is really just a placeholder.) And whatever valid version I put there, will of course be overridden by the entry in `override_targets`. So the specified version is either misleading or needs to be kept manually in sync with whatever pulled in `@com.google.protobuf:protobuf-java`.

So assuming Issue #273 is resolved, this will work (fsvo "work"), and I am thankful for the suggestion.
